robot/impl/local_robot.go

```diff
@@ -45,6 +45,9 @@ type localRobot struct {
 	manager *resourceManager
 	config  *config.Config
 
+	fsConfig    *framesystem.Config
+	lastFSError error
```

Comment:
it jumps out to me as a little weird to be storing this in the robot struct.  open to other ideas

Reply:
it seems like framesystem service can hold this

Reply:
Yeah I toyed with this, but some of the errors we care about currently exist in the robot file (i.e. `errors.Errorf("cannot give frame system part the name %s", referenceframe.World)`

A way around this would be removing all checks from the robot level and passing in whatever config we get for the frame system service to handle.  This is probably the better approach I'll look into this

Reply:
Agree this is the right approach here